src/c4/C4_MPI.i.hh

```diff
@@ -0,0 +1,98 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   c4/C4_MPI.i.hh
+ * \author Alex R Long
+ * \date   Mon Aug 21 07:47:01 2017
+ * \brief  C4 MPI standard implementations.
+ * \note   Copyright (C) 2017 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef c4_C4_MPI_i_hh
+#define c4_C4_MPI_i_hh
+
+#ifdef C4_MPI
+
+namespace rtt_c4 {
+
+//---------------------------------------------------------------------------//
+template <typename T>
+DLL_PUBLIC_c4 void send_is_custom(C4_Req &request, const T *buffer, int size,
+                                  int destination, int tag) {
+
+  Require(!request.inuse());
+
+  // set the request
+  request.set();
+
+  Remember(int const retval =)
+      MPI_Issend(const_cast<T *>(buffer), size, T::MPI_Type, destination, tag,
+                 communicator, &request.r());
+  Check(retval == MPI_SUCCESS);
+
+  return;
+}
+
+//---------------------------------------------------------------------------//
+template <typename T>
+DLL_PUBLIC_c4 int send_custom(const T *buffer, int size, int destination,
+                              int tag) {
+  MPI_Send(const_cast<T *>(buffer), size, T::MPI_Type, destination, tag,
+           communicator);
+  return C4_SUCCESS;
+}
+
```

Comment:
Please add a unit test for `int send_custom`, `int receive_custom` and `int message_size_custom`.